packages-exp/auth-exp/test/integration/webdriver/persistence.test.ts

```diff
@@ -139,9 +158,84 @@ browserDescribe('WebDriver persistence test', driver => {
       await driver.waitForAuthInit();
       expect(await driver.getUserSnapshot()).to.equal(null);
     });
-  });
 
-  // TODO: Upgrade tests (e.g. migrate user from localStorage to indexedDB).
+    it('migrate stored user from localStorage if indexedDB is available', async () => {
+      const persistedUser = await testPersistedUser();
+      await driver.webDriver.navigate().refresh();
+      await driver.call(PersistenceFunction.LOCAL_STORAGE_SET, {
+        [fullPersistenceKey]: persistedUser
+      });
+      await driver.injectConfigAndInitAuth();
+      await driver.waitForAuthInit();
+
+      // User from localStorage should be picked up.
+      const user = await driver.call<User>(CoreFunction.USER_SNAPSHOT);
```

Comment:
why not `driver.getUserSnapshot()`? (Comes pre-typed)

Reply:
Done!